### PR TITLE
x86: Enable batch clearing by default

### PR DESCRIPTION
**Pull request by BradleyWood**
This change needs to be tested with eclipse/omr#7234

### CODE REVIEW

**Comment by vijaysun-omr** [comment]
I would appreciate a review from @0xdaryl on this to make sure we have considered what we need to, before making this change

**Comment by dmitripivkine** [comment]
Side note (you aware about this probably): non-zeroed TLH can be used for allocation of primitive arrays only. And yes, for so many years we can not achieve perf win of pre-zeroing TLH in x86, so it was not enabled.

**Comment by dmitripivkine** [comment]
> Don't we also need to switch on `-Xgc:batchClearTLH` by default to properly enable this feature ?

Everything should work if this line in `DLLMain.cpp` is called eventually:
```
vm->memoryManagerFunctions->allocateZeroedTLHPages(vm, true);
```
And seems this change does this,

**Review by vijaysun-omr — APPROVED** [review]

**Comment by vijaysun-omr** [comment]
I am only approving the fact that the enablement looks okay. I'll wait for Daryl's review and run tests.

**Comment by dmitripivkine** [comment]
Another side question is can memory zeroing performance be improved by customizing `OMRZeroMemory()` for x86. Current default implementation rely on standard memset().

**Comment by BradleyWood** [comment]
> Another side question is can memory zeroing performance be improved by customizing OMRZeroMemory() for x86. Current default implementation rely on standard memset().

I am experimenting with that

**Comment by dmitripivkine** [comment]
> > Another side question is can memory zeroing performance be improved by customizing OMRZeroMemory() for x86. Current default implementation rely on standard memset().
> 
> I am experimenting with that

Please note that de facto OMRZeroMemory() should have start address/size alined to processor word size. This is natural limitation from other platforms implementations. It might simplify customization task a little bit.

**Comment by 0xdaryl** [comment]
I think the code paths that need to be aware of dual TLH are aware of it.

Can you describe how you tested this?  Presumably on systems and workloads within IBM.  Since you enable this for 32-bit as well this should include tests on 32-bit Java 8.

> I am experimenting with that
Please report your conclusions from that.  I suspect memset() performance is quite good, but would like to hear your conclusion.

File: runtime/compiler/control/DLLMain.cpp [hunk 1]
```diff
@@ -357,7 +357,7 @@ IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void * reserved)
             /* Platform detection seems difficult at this point.                              */
             {
 #ifdef TR_HOST_X86
-            static char *enableBatchClear = feGetEnv2("TR_EnableBatchClear", (void *)vm);
+            static char *disableBatchClear = feGetEnv2("TR_DisableBatchClear", (void *)vm);
```
Comment (0xdaryl) [inline runtime/compiler/control/DLLMain.cpp:360]: Why not pull this out of the `#ifdef` as part of the cleanup?  It is the same on both paths of the ifdef.

**Comment by BradleyWood** [comment]
> We may need to consider some architecture version check (e.g. >= skylake)

The port library doesn't support this yet, but its being worked on.

**Pull request closed without merging**